mqr/utils.py

```diff
@@ -70,3 +71,49 @@ def get_next_send_date():
     # TODO: maybe this should be smarter
     # calculate date based on edd_or_dob and next week, incase message failed?
     return get_today() + timedelta(weeks=1)
+
+
+def get_facility_province(facility_code):
+    try:
+        clinic_code = ClinicCode.objects.get(code=facility_code)
+    except ClinicCode.DoesNotExist:
+        clinic_code = None
+
+    return clinic_code
+
+
+def get_weeks_pregnant(edd):
+    full_term_weeks = 40
+    estimated_date = edd
+
+    if isinstance(edd, str):
+        # convert to date
+        estimated_date = datetime.strptime(edd, "%Y-%m-%d").date()
+    elif isinstance(edd, datetime):
+        estimated_date = edd.date()
+
+    # Get remaining weeks
+    remaining_weeks = (estimated_date - get_today()).days // 7
+
+    weeks_pregnant = full_term_weeks - remaining_weeks
+
+    if 16 <= weeks_pregnant <= 20:
+        return "16-20"
+    elif 21 <= weeks_pregnant <= 25:
+        return "21-25"
+    elif 26 <= weeks_pregnant <= 30:
+        return "26-30"
+    else:
+        return None
+
+
+def get_age_bucket(age):
+    mom_age = int(age)
```

Comment:
I think we can also make the serializer field for this an int, then we won't need this

Reply:
Done